### ISSUE TITLE
Add ability to specify option may only be specified once

### ISSUE THREAD

**Issue by plroebuck** [yargs/yargs#465]
Add boolean method `.once(<key>)` to disallow specification of same option more than once.
Default value is `false`.

Given single numeric argument `x`, throw if `.once('x')` was set with given command line:
```bash
$ my --x 5 --arg 8
```

If applied to boolean argument, it would additionally throw if given both `--bool` and `--no-bool`.

**Comment by timoxley** [comment]
If it even just picked the last specified value, that would be good. 

As far as I can tell, every single arg might be the type that you configured, or it might be an array of that type, there's no way to force one or the other, so to be safe you kinda have to check all of them. Ugh.

This quirk bit me hard recently, was trying to override a flag set in an npm script from the command-line:

e.g. `npm start` script says something like:

```json
{
  "scripts": {
    "start: "node app.js --interval=1000"
  }
}
```

and I want to override this from the cli:
```
> npm start -- --interval=10000
```

Inside the app, the value of the `--interval` argument is being passed as the `delay` parameter in a `setInterval(fn, delay)`. In this case, instead of getting either the original value `1000`, or the desired value `10000` the `setInterval` receives both as an array of `[1000, 10000]. This ends up being equivalent to `setInterval(() => {}, 0)`, which sends the application crazy.

**Comment by bcoe** [comment]
@plroebuck @timoxley could we start by introducing a new configuration option to `yargs-parser` that disables array expansion?

https://github.com/yargs/yargs-parser#configuration

or do you require both types of behavior in the same app?

**Comment by timoxley** [comment]
@bcoe yeah in my case, both. Some things are set up for both scenarios, others not.

I was thinking something like `.option('interval', {array: false})` could be a good way to signal it perhaps.

**Comment by timoxley** [comment]
or the original proposed `.once('interval')`

**Comment by tnunes** [comment]
Any updates on this?

I'm facing the same issue, where I have an option that must be a single value, but if specified multiple times, yargs will give me an array of values.

I'm good with either `{ array: false }` or `{ once: true }`, as long as we have the feature :)

**Comment by plroebuck** [comment]
I'd prefer the { once:true } annotation, but means same thing in @timoxley's case above.

**Comment by plroebuck** [comment]
Based on `README.md`, I wanted to throw for both of these cases:

**duplicates**
```bash
$ node examples/reflect.js -x 5 -x 8 -x 0
```

**negate fields** (slightly modified)
```bash
$ node examples/reflect.js -b --no-b
```

Once a (non-default) value is assigned to an option, it may not be reassigned later
(from client perspective). Partial implementation that disabled array expansion
[which I view as subset of my proposal] would be an acceptable start; the duplicate
issue is obviously materially affecting others.

**Comment by plroebuck** [comment]
@bcoe, any progress?

**Comment by plroebuck** [comment]
@bcoe, any progress?

**Comment by bcoe** [comment]
@plroebuck there's now the option `duplicate-arguments-array` which can be set to `false` with `yargs.parserConfiguration`, this will make the parser use the last value provided for `-x`, rather than coercing into an array.

I think given that folks probably want this behavior across the board for their application, this feature is probably sufficient.

**Comment by shadowspawn** [comment]
This issue has not had any activity in five years.

Feel free to open a new issue if it comes up again, with new information and renewed interest.
 
Thank you for your contributions.
